synopsys-task/src/main.ts

```diff
@@ -72,19 +83,27 @@ export async function run() {
   console.log("Synopsys Task workflow execution completed");
 }
 
-export function logBridgeExitCodes(message: string): string {
-  const exitCode = message.trim().slice(-1);
+export function logBridgeExitCodes(message: string, exitCode: string): string {
   return constants.EXIT_CODE_MAP.has(exitCode)
     ? "Exit Code: " + exitCode + " " + constants.EXIT_CODE_MAP.get(exitCode)
-    : message;
+    : "Undefined error from extension: " + message + constants.SPACE + "999";
```

Comment:
Use constants instead of direct error codes so that later codes can be changed from one single source.
Example, UNDEFINED_ERROR_CODE = 999

Follow this in all the places.